File: GeneratedEntityReaderClasses/Components/AudioComponent.cs
namespace GeneratedNoitaClasses;
public class AudioComponent : NoitaComponentBase,  iNoitaType<AudioComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString file { get; set; } = new();
	public NoitaString event_root { get; set; } = new();
	public NoitaString audio_physics_material { get; set; } = new();
	public NoitaBool set_latest_event_position { get; set; } = new();
	public NoitaBool remove_latest_event_on_destroyed { get; set; } = new();
	public NoitaBool send_message_on_event_dead { get; set; } = new();
	public NoitaBool play_only_if_visible { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		file.Read(s);
		event_root.Read(s);
		audio_physics_material.Read(s);
		set_latest_event_position.Read(s);
		remove_latest_event_on_destroyed.Read(s);
		send_message_on_event_dead.Read(s);
		play_only_if_visible.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		file.Write(s);
		event_root.Write(s);
		audio_physics_material.Write(s);
		set_latest_event_position.Write(s);
		remove_latest_event_on_destroyed.Write(s);
		send_message_on_event_dead.Write(s);
		play_only_if_visible.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/AudioComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString

from NoitaStream import NoitaStream
@dataclass
class AudioComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    file:NoitaString = field(default_factory=NoitaString)
    event_root:NoitaString = field(default_factory=NoitaString)
    audio_physics_material:NoitaString = field(default_factory=NoitaString)
    set_latest_event_position:NoitaBool = field(default_factory=NoitaBool)
    remove_latest_event_on_destroyed:NoitaBool = field(default_factory=NoitaBool)
    send_message_on_event_dead:NoitaBool = field(default_factory=NoitaBool)
    play_only_if_visible:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.file.Read(s)
        self.event_root.Read(s)
        self.audio_physics_material.Read(s)
        self.set_latest_event_position.Read(s)
        self.remove_latest_event_on_destroyed.Read(s)
        self.send_message_on_event_dead.Read(s)
        self.play_only_if_visible.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.file.Write(s)
        self.event_root.Write(s)
        self.audio_physics_material.Write(s)
        self.set_latest_event_position.Write(s)
        self.remove_latest_event_on_destroyed.Write(s)
        self.send_message_on_event_dead.Write(s)
        self.play_only_if_visible.Write(s)

